Applications/Utils/MeshEdit/removeMeshElements.cpp

```diff
@@ -28,26 +28,46 @@
 #include "MeshLib/MeshEditing/RemoveMeshComponents.h"
 
 template <typename PROPERTY_TYPE>
-void searchByProperty(std::string const& property_name,
-                      std::vector<PROPERTY_TYPE> const& property_values,
-                      MeshLib::ElementSearch& searcher)
+void searchByPropertyValue(std::string const& property_name,
+                           std::vector<PROPERTY_TYPE> const& property_values,
+                           MeshLib::ElementSearch& searcher)
 {
-    for (auto const& property_value : property_values) {
-        const std::size_t n_marked_elements =
-            searcher.searchByPropertyValue(property_value, property_name);
+    for (auto const& property_value : property_values)
+    {
+        std::size_t n_marked_elements = searcher.searchByPropertyValue<double>(
+            property_name, property_value);
+        if (n_marked_elements == 0)
+            std::size_t n_marked_elements = searcher.searchByPropertyValue<int>(
```

Comment:
```searchByPropertyValue``` just calls ```searchByPropertyValueRange```. I've just put this here for backwards compatibility because searching for a specific value was used in a few tests IIRC.

Additional types would need to be implemented her by adding a line with the respective type. The search-method itself just checks if the value for each element is smaller/larger than the range bounds. We just need to test the type because the property vector won't be returned without knowing the correct type.

The order of type-tests shouldn't matter because only one of them can ever return a nonzero value. All the wrong types aren't tested at all because they return a nullptr (or rather a boost optional, but it's essentially the same result) and thus will automatically return zero.